sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -91,18 +92,28 @@ def get_historical_features(
         expected_join_keys = _get_join_keys(project, feature_views, registry)
 
         if type(entity_df) is str:
-            entity_df_job = client.query(entity_df)
-            # Start the job and get the schema back. We don't need the actual rows.
-            entity_df_result = entity_df_job.result(max_results=0)
+            entity_df_job: QueryJob = client.query(entity_df)
+            # Note: Don't call entity_df_job.result() here, it will lead
+            # to OOM errors. Instead just wait until the job is done.
+            while not entity_df_job.done():
+                time.sleep(1)
```

Comment:
`done()` has a timeout and retry btw.

Reply:
Didn't realize that, but with the new approach won't be using done(), so I'm resolving this.